staging/src/k8s.io/apimachinery/pkg/api/equality/semantic.go

```diff
@@ -46,4 +47,25 @@ var Semantic = conversion.EqualitiesOrDie(
 	func(a, b fields.Selector) bool {
 		return a.String() == b.String()
 	},
+	func(a, b metav1.LabelSelectorRequirement) bool {
```

Comment:
I don't think we should do this to fix a test, the implications of this are much bigger than just the test

Reply:
changing the equality definition will allow mutation of objects containing a selector which today are verified to be immutable on update

also, it didn't sound like our parser / stringifier correctly round-tripped these... papering over that by acting like they are equal isn't the right fix